raiden/network/transport/matrix.py

```diff
@@ -502,40 +572,42 @@ def _handle_message(self, room, event):
                     peer_address=peer_address,
                 )
                 return
-            self._receive_message(message)
+            if isinstance(message, Delivered):
+                self._receive_delivered(message)
+            else:
+                self._receive_message(message)
         else:
             self.log.error(
                 'Invalid message',
                 message=data,
             )
 
     def _receive_delivered(self, delivered: Delivered):
-        # FIXME: The signature doesn't seem to be verified - check in UDPTransport as well
+        # FIXME: check if UDPTransport also checks Delivered sender and message presence
```

Comment:
The sender must be checked, otherwise a malicious node could forge a delivered message, which would clear the queue and stop retries.

This is only necessary because UDP is *not* an encrypted transport, so we need to authenticate using signatures.

Reply:
Yes, I know.. the point is that I did that there for `Matrix`, so it's just updating the `FIXME` to remember to do this check also for `UDP` eventually.